src/http_client/base.ts

```diff
@@ -65,10 +65,18 @@ export class ApiRequest {
     try {
       const response = await fetch(target, options);
       let responseJSON;
-      if (response.status === 204) {
-        responseJSON = null;
-      } else {
-        responseJSON = await response.json();
+
+      try {
+        if (response.status === 204) {
```

Comment:
can you write a test for this? you can use msw to intercept the request

Reply:
@kibertoad Sure!